### PR TITLE
feat(types): add entrypoint 0.7 contract types

### PR DESCRIPTION
**Pull request by dancoombs**
## Proposed Changes

  - Compiles both versions of the entry point contracts
  - Only consumes v0.6 for now, going to begin abstracting these in subsequent PRs

### CODE REVIEW

File: crates/types/build.rs [hunk 1]
```diff
@@ -19,48 +19,146 @@ fn main() -> Result<(), Box<dyn error::Error>> {
...
 }
 
-fn abigen_of(contract: &str) -> Result<Abigen, Box<dyn error::Error>> {
-    Ok(Abigen::new(
-        contract,
-        format!("contracts/out/{contract}.sol/{contract}.json"),
-    )?)
+fn generate_v0_7_bindings() -> Result<(), Box<dyn error::Error>> {
+    run_command(
+        Command::new("forge")
```
Comment (0xfourzerofour) [inline crates/types/build.rs:65]: Nit: there is some repeated logic here, you could make the forge script more configurable so that you pass the v_6, v_7 and utils into the same function to build the command dynamically

**Review by 0xfourzerofour — COMMENTED** [review]
small nit about code reuse

File: crates/types/contracts/test/PrecompileAccountTest.sol [hunk 2]
```diff
@@ -1,8 +1,8 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.14;
 
-import "../src/PrecompileAccount.sol";
-import "account-abstraction/interfaces/UserOperation.sol";
+import "../src/v0_6/PrecompileAccount.sol";
```
Comment (alex-miao) [inline crates/types/contracts/test/PrecompileAccountTest.sol:4]: Should tests also be split into 0.6/0.7

Comment (dancoombs) [reply crates/types/contracts/test/PrecompileAccountTest.sol:4]: I tried to clean this up as much as I could in follow up PRs. We do need to use explicit types during tests, and it doesn't make sense to duplicate every test for v0.7. We might want to come up with a better approach for this going forward. But for now, going to mostly rely on the shared code on traits and specific tests for v0.7 types.

File: crates/types/build.rs (hunk 1 above)
Comment (dancoombs) [reply crates/types/build.rs:65]: Good idea, updated!
